ringpop.go

```diff
@@ -164,25 +171,29 @@ func (rp *Ringpop) identity() (string, error) {
 // TChannel object on the Ringpop instance.
 func (rp *Ringpop) channelIdentityResolver() (string, error) {
 	hostport := rp.channel.PeerInfo().HostPort
+	// Check that TChannel is listening. By default, TChannel listens on an
+	// ephemeral host/port. The real port is then assigned by the OS when
+	// ListenAndServe is called. If the hostport is 0.0.0.0:0, it means
+	// TChannel is not yet listening and the hostport cannot be resolved.
+	if hostport == "0.0.0.0:0" {
+		return "", fmt.Errorf("unable to resolve valid listen address (TChannel hostport is %s)", hostport)
+	}
 	return hostport, nil
 }
 
 // Destroy Ringpop
 func (rp *Ringpop) Destroy() {
-	rp.node.Destroy()
 
-	rp.state.Lock()
-	rp.state.destroyed = true
-	rp.state.Unlock()
-}
+	if rp.node != nil {
```

Comment:
Should we add a condition above?

```
if rp.getState() == destroyed {
  return;
}
```

I feel like this'd support the claim that this operation were idempotent better. Otherwise, you'd then be dependent upon whether `rp.node.Destroy()` were idempotent too.